File: androidApp/app/src/main/java/com/example/productinfoapp/data/model/Nutrition.java
package com.example.productinfoapp.data.model;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;

public class Nutrition implements Serializable {
    @SerializedName("calories")
    private double calories;

    @SerializedName("sugar")
    private double sugar;

    @SerializedName("fat")
    private double fat;

    @SerializedName("saturatedFat")
    private double saturatedFat;

    @SerializedName("salt")
    private double salt;

    @SerializedName("protein")
    private double protein;

    @SerializedName("fiber")
    private double fiber;

    @SerializedName("servingSize")
    private double servingSize;

    @SerializedName("servingUnit")
    private String servingUnit;

    public Nutrition() {}

    // Getters
    public double getCalories() { return calories; }
    public double getSugar() { return sugar; }
    public double getFat() { return fat; }
    public double getSaturatedFat() { return saturatedFat; }
    public double getSalt() { return salt; }
    public double getProtein() { return protein; }
    public double getFiber() { return fiber; }
    public double getServingSize() { return servingSize; }
    public String getServingUnit() { return servingUnit; }

    // Setters (required by Room @Embedded)
    public void setCalories(double calories) { this.calories = calories; }
    public void setSugar(double sugar) { this.sugar = sugar; }
    public void setFat(double fat) { this.fat = fat; }
    public void setSaturatedFat(double saturatedFat) { this.saturatedFat = saturatedFat; }
    public void setSalt(double salt) { this.salt = salt; }
    public void setProtein(double protein) { this.protein = protein; }
    public void setFiber(double fiber) { this.fiber = fiber; }
    public void setServingSize(double servingSize) { this.servingSize = servingSize; }
    public void setServingUnit(String servingUnit) { this.servingUnit = servingUnit; }
}

File: backend/models/Nutrition.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Nutrition = sequelize.define('Nutrition', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  calories: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  sugar: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  fat: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  saturatedFat: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  salt: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  protein: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  fiber: {
    type: DataTypes.FLOAT,
    defaultValue: 0
  },
  servingSize: {
    type: DataTypes.FLOAT,
    defaultValue: 100
  },
  servingUnit: {
    type: DataTypes.STRING(20),
    defaultValue: 'g'
  }
}, {
  timestamps: false
});

module.exports = Nutrition;

